### ISSUE TITLE
snakemake as an HPC backend?

### ISSUE THREAD

**Issue by wlandau** [ropensci/drake#368]
I have never actually used [`snakemake`](https://snakemake.readthedocs.io/en/stable/), but it looks incredibly deep, thorough, and sophisticated, definitely the sort of tool I should have learned about before starting `drake`. I am especially interested in the scheduling algorithm (slides 16 through 20 in [this presentation](http://slides.com/johanneskoester/deck-1) by @johanneskoester). It will be a long time before we have proper native scheduling and load balancing for all of `drake`'s HPC backends, and I think a `make(parallelism = "snakemake")` capability could grant users a ton of parallel efficiency in the meantime. I have a lot to learn about `snakefile`s, but ultimately I think the implementation will be analogous to `make(parallelism = "Makefile")`.

**Comment by wlandau** [comment]
With #407 well underway, I will postpone this one until we hit another wall in development.
